sc/src/main/java/uk/ac/ebi/atlas/experimentpage/JsonExperimentTSnePlotController.java

```diff
@@ -0,0 +1,162 @@
+package uk.ac.ebi.atlas.experimentpage;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import uk.ac.ebi.atlas.experimentpage.baseline.tsne.TSnePlotService;
+import uk.ac.ebi.atlas.experimentpage.baseline.tsne.TSnePoint;
+import uk.ac.ebi.atlas.experimentpage.json.JsonExperimentController;
+import uk.ac.ebi.atlas.model.experiment.Experiment;
+import uk.ac.ebi.atlas.trader.ScxaExperimentTrader;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.ac.ebi.atlas.utils.GsonProvider.GSON;
+
+@RestController
+public class JsonExperimentTSnePlotController extends JsonExperimentController {
+    private final TSnePlotService tSnePlotService;
+
+    @Inject
+    public JsonExperimentTSnePlotController(ScxaExperimentTrader experimentTrader, TSnePlotService tSnePlotService) {
+        super(experimentTrader);
+        this.tSnePlotService = tSnePlotService;
+    }
+
+    @RequestMapping(
+            value = "/json/experiments/{experimentAccession}/tsneplot/{perplexity}/clusters/k/{k}",
```

Comment:
Is the idea here to have a `/clusters/factors/{factorId}` type of endpoint as well?

Reply:
Yes it is. ;)